src/query/functions/fetch.go

```diff
@@ -103,23 +106,32 @@ func (n *FetchNode) fetch(queryCtx *models.QueryContext) (block.Result, error) {
 	sp, ctx := opentracing.StartSpanFromContext(ctx, "fetch")
 	defer sp.Finish()
 
-	timeSpec := n.timespec
-	// No need to adjust start and ends since physical plan
-	// already considers the offset, range
-	startTime := timeSpec.Start
-	endTime := timeSpec.End
+	var (
+		timeSpec = n.timespec
+		// No need to adjust start and ends since physical plan already considers
+		// the offset, range.
+		startTime = timeSpec.Start
+		endTime   = timeSpec.End
+		step      = timeSpec.Step
+		offset    = n.op.Offset
+	)
+
+	if n.overrideOptions.ShouldOverride {
+		step = n.overrideOptions.Step
+		startTime = startTime.Add(-1 * n.overrideOptions.StartOffset)
```

Comment:
Hah lovely, we're doing offsets again. Good times. 

Will try to deep dive into this, hopefully by avoiding using a context to flow the offset it is cleaner than M3QL impl?